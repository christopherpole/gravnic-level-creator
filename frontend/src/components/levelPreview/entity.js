import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Tile from '../common/tile';
import { GRID_SIZE } from '../../config/settings';

export const Wrapper = styled.div`
  height: ${props => props.heightPercentage}%;
  width: ${props => props.widthPercentage}%;
  position: absolute;
  left: ${props => props.xPos}%;
  top: ${props => props.yPos}%;
`;

export const Entity = ({ entityId, xPos, yPos }) => (
  <Wrapper
    xPos={xPos}
    yPos={yPos}
    heightPercentage={100 / GRID_SIZE}
    widthPercentage={100 / GRID_SIZE}
  >
    <Tile tileId={entityId} />
  </Wrapper>
);

Entity.propTypes = {
  entityId: PropTypes.number.isRequired,
  xPos: PropTypes.number.isRequired,
  yPos: PropTypes.number.isRequired,
};

export default Entity;
