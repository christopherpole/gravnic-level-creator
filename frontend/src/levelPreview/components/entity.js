import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { GRID_SIZE } from 'config/settings';
import { Tile } from 'common/tiles';

export const Wrapper = styled.div`
  height: ${props => props.heightPercentage}%;
  width: ${props => props.widthPercentage}%;
  position: absolute;
  left: ${props => props.xPos}%;
  top: ${props => props.yPos}%;
  transition: all ${props => props.moveSpeed}ms linear;
  z-index: ${props => (props.isMovableEntity ? 1 : 0)};
`;

export const Entity = ({ entity, xPos, yPos, gameSpeed, isMovableEntity }) => (
  <Wrapper
    xPos={xPos}
    yPos={yPos}
    heightPercentage={100 / GRID_SIZE}
    widthPercentage={100 / GRID_SIZE}
    moveSpeed={gameSpeed}
    isMovableEntity={isMovableEntity}
  >
    <Tile entity={entity} moveSpeed={gameSpeed} />
  </Wrapper>
);

Entity.propTypes = {
  xPos: PropTypes.number.isRequired,
  yPos: PropTypes.number.isRequired,
  gameSpeed: PropTypes.number.isRequired,
  isMovableEntity: PropTypes.bool.isRequired,
  entity: PropTypes.shape({
    entityId: PropTypes.string.isRequired,
    fading: PropTypes.bool,
    color: PropTypes.string,
    targetEntity: PropTypes.object,
    shrinking: PropTypes.bool,
  }).isRequired,
};

const mapStateToProps = state => ({
  gameSpeed: state.levelPreview.gameSpeed,
});

export default connect(mapStateToProps)(Entity);
