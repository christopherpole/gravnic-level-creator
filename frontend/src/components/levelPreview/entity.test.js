import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { Entity } from './entity';

configure({ adapter: new Adapter() });

describe('Level preview entities', () => {
  let props;

  beforeEach(() => {
    props = {
      entityId: 3,
      xPos: 10,
      yPos: 20,
      gameSpeed: 100,
    };
  });

  it('Renders without exploding', () => {
    const entity = shallow(<Entity {...props} />);

    expect(entity).toHaveLength(1);
  });

  it('Matches the current snapshot', () => {
    const entity = shallow(<Entity {...props} />);

    expect(toJson(entity)).toMatchSnapshot();
  });
});
