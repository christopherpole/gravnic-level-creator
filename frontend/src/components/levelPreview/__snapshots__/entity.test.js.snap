// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level preview entities Matches the current snapshot 1`] = `
<styled.div
  entityId={3}
  fading={false}
  heightPercentage={10}
  moveSpeed={100}
  widthPercentage={10}
  xPos={10}
  yPos={20}
/>
`;
